aws_lambda_builders/workflows/java_gradle/resources/lambda-build-init.gradle

```diff
@@ -0,0 +1,103 @@
+import groovy.io.FileType
+import java.nio.file.FileAlreadyExistsException
+import java.nio.file.Files
+import java.nio.file.FileVisitOption
+import java.nio.file.FileVisitResult
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.SimpleFileVisitor
+import java.security.MessageDigest
+import java.util.zip.ZipInputStream
+import java.util.zip.ZipOutputStream
+import java.util.zip.ZipEntry
+
+gradle.ext.SCRATCH_DIR_PROPERTY = 'software.amazon.aws.lambdabuilders.scratch-dir'
+
+// Called after the project has been evaluated
+gradle.afterProject({p ->
+    def buildDir = buildDirForProject(p)
+    p.buildDir = buildDir
+})
+
+gradle.taskGraph.afterTask{ t ->
+    if (t.name != 'build') {
+        return;
+    }
+
+    assertExpectedBuildDir(t.project)
+
+    def artifactJars = t.project.configurations.archives.artifacts.files.files
+
+    def runtimeCpJars = t.project.configurations.runtimeClasspath.files
+
+    def zipArtifact = createArtifactDir(t.project).resolve("${t.project.name}.zip")
```

Comment:
Actually, do you really need to create a zip here? Can you instead write everything into a directory that SAM CLI will eventually zip everything up using `sam package`? 

In terms of workflow, `sam build` -> `sam local invoke` -> `sam package` -> `sam deploy`, writing a zip here means the `local invoke` will unzip and rezip eventually during `package`.

Reply:
Hmm but where do we draw the line between SAM and the builders? If customers want to just use the builders library, it would be more convenient if they didn't need to ZIP if themselves

Reply:
The scope of builder is to "compile" the code into deployable artifacts. How you package the deployable artifacts is the responsibility of a packager (which currently is in SAM CLI). It is zip if you want to deploy to Lambda, but we could also package artifacts for say publishing to Serverless Apps Repo which would be a different format. We might package each function separately or all functions together, but those are decisions the packager should make.

Reply:
Makes sense. Do I just treat `artifacts_dir` as if it were the root of the ZIP? e.g.

```
artifact_dir/com/amazonaws/Foo.class
artifact_dir/com.amazonaws/Bar.class
lib/dependency1.jar
```
?